src/utils.ts

```diff
@@ -1,11 +1,14 @@
-export function validatePhoneOrEmail(str: string) {
-  const regex = /^(?:(?:\+?\d{1,3}[- ]?)?(?:\(\d{2,5}\)|\d{2,5})[- ]?)?(?:\d{1,4}[- ]?)?\d{1,10}$|^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,4}$/;
+const emailRegex = /^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,4}$/;
+const phoneRegex =
+  /^(?:\+?\d{1,3}[-.\s]?)?(?:\(\d{2,5}\)|\d{2,5})[-.\s]?\d{1,4}[-.\s]?\d{1,4}[-.\s]?\d{1,9}$/;
```

Comment:
Comment: Phone validation isn't the best thing to ask in an interview because it's complicated to get all the cases in the world, but there's plenty of libraries for it, like this one: 

https://www.npmjs.com/package/phone?activeTab=readme

Always good to check bundle size though

https://bundlephobia.com/package/phone@3.1.40

I would consider this "alright", but it depends on what other libraries you need for your web app.